src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -126,16 +128,16 @@ public function getIriFromItem($item, int $referenceType = UrlGeneratorInterface
             ), $e->getCode(), $e);
         }
 
-        return $this->getItemIriFromResourceClass($resourceClass, $identifiers, $referenceType);
+        return $this->getItemIriFromResourceClass($resourceClass, $identifiers, UrlGeneratorInterface::DEFAULT === $referenceType ? $this->routerStrategyType : $referenceType);
```

Comment:
By using null, you’ll also be able to simplify these check apps by using the coalesce operator.